### ISSUE TITLE
ewallet docker container can't start

### ISSUE THREAD

**Issue by kartsims** [omisego/ewallet#598]
# Subject of the issue

When starting a fresh instance of the ewallet, using `docker-compose`, the `ewallet` container keeps restarting.

# Your environment

OSX Mojave

```
~ ■ docker -v
Docker version 18.09.0, build 4d60db4

~ ■ docker-compose -v
docker-compose version 1.23.2, build 1110ad01
```

# Steps to reproduce

All information here : https://gist.github.com/kartsims/3514ed2cdab4abd9b28c46ae9c41ae22

After [discussion](https://gist.github.com/kartsims/3514ed2cdab4abd9b28c46ae9c41ae22#gistcomment-2787354) with @unnawut it appears that since the customizable settings are stored in the DB, the ewallet app makes a query in the DB, which fails, and triggers a restart on the container.

In this context, the seed command can never be executed.

On a sidenote, the docker command proposed to seed could be updated to leverage docker-compose's environment and links as such :

```
docker-compose run ewallet bash -c "env MIX_ENV=prod mix do local.hex --force, local.rebar --force, ecto.create, ecto.migrate, seed --sample"
```

_Note that it is using `run` so we can seed the ewallet before starting the container using `docker-compose up`_

**Comment by kartsims** [comment]
Nice! Thanks @sirn 

May I suggest to add explicit [container names](https://docs.docker.com/compose/compose-file/#container_name) too ? Using default container names (that get suffixed with a number and prefixed with the directory's name) isn't very elegant, and completely useless in a dev environment.

Also, this prevents me from releasing the JS lib for the ewallet API that I am working on (not that the whole world is waiting for it... 😆).

**Comment by unnawut** [comment]
Just a side note that kartsim's `docker-compose run ...` command worked well for me.

A slight addition so that the runned container doesn't experience the restart loop the same way as the main container by disabling the entrypoint:

```
$ docker-compose run --entrypoint="" ewallet bash -c "env MIX_ENV=prod mix do local.hex --force, local.rebar --force, ecto.create, ecto.migrate, seed --sample"
```

**Comment by sirn** [comment]
The new `docker-compose.yml` is already landed. In the meantime, please check out [docker-compose.yml](https://github.com/omisego/ewallet/blob/master/docker-compose.yml) in the master branch. You might also found `docker-gen.sh` useful too (which should auto-configure everything):

```
$ ./docker-gen.sh > docker-compose.override.yml
$ docker-compose run --rm ewallet bin/ewallet initdb
$ docker-compose run --rm ewallet bin/ewallet seed
$ docker-compose up
```

By default the `docker-compose.yml` uses the new `omisego/ewallet:dev` image which tracks master. :-) Documentation is non-existent at the moment but I'm working on it as well as supporting container names.

**Comment by samundra** [comment]
@sirn I think it's better to include these instructions in the documentation. I had to fiddle around a lot with docker-compose and had to retry seeding command multiple times to actually get seeding to work.

I also got a problem with dev image `omisego/ewallet:dev`. I got the following error while executing command.

``` bash
$ docker exec -it $(docker ps --filter ancestor="omisego/ewallet:dev" --format "{{.Names}}") \
  env MIX_ENV=prod mix do \
  local.hex --force, local.rebar --force, \
  ecto.create, ecto.migrate, seed --sample

[OUTPUT]
env: can't execute 'mix': No such file or directory
```

To solve it, I switched to another image 'omisego/ewallet:latest`. The previous error was gone but then I ended up with database seeding problem. To solve seeding problem, I had to execute the above command multiple times and just waited to get lucky 😄  so that it finishes migrating before container restarts with 🤞 

I then tested by trying to access `http://localhost:4000` which returned `{"status":true,"ewallet_version":"1.1.0"}`
